src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvConnection.cs

```diff
@@ -100,6 +102,16 @@ public LibuvConnection()
             }
         }
 
+        public override void OnApplicationComplete()
+        {
+            Thread.Post(connection =>
+            {
+                // Call uv_close which will fire any pending uv_write callbacks
+                connection._socket.Dispose();
```

Comment:
> EDIT: This needs a bunch of work. It's too easy to cut valid responses off.

This here is why. Since WriteAsync is a lie, the app can complete well before the socket can be closed.